calyx/src/passes/component_iniliner.rs

```diff
@@ -457,7 +457,8 @@ impl Visitor for ComponentInliner {
         // If the associated instance has been inlined, replace the invoke with
         // its control program.
         let cell = s.comp.borrow();
-        if let Some(con) = self.control_map.get(cell.name()) {
+        if let Some(con) = self.control_map.get_mut(cell.name()) {
+            con.get_mut_attributes().insert("new_fsm", 1);
```

Comment:
Can we put this behavior behind a flag? Not sure if we want this to be a default